concepts/05 UI Components/DataGrid/10 Enhance Performance on Large Datasets/030 Rendering Optimization.md

```diff
@@ -1,7 +1,16 @@
-The UI component renders all rows and columns once the data is loaded. To improve performance, set the [rowRenderingMode](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/scrolling/rowRenderingMode.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/scrolling/#rowRenderingMode') and [columnRenderingMode](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/scrolling/columnRenderingMode.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/scrolling/#columnRenderingMode') properties in the **scrolling** object to *"virtual"*. These properties specify that UI elements are only rendered when they come into the viewport.
+If you work with a large data set, apply the following recommendations to optimize rendering performance:
 
-Use virtual row rendering when [paging is disabled](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/paging/enabled.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/paging/#enabled') or the [pageSize](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/paging/pageSize.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/paging/#pageSize') is more than the default value.
-
-Virtual column rendering can be used when there are more than 20 columns outside the viewport.
+* **Render only visible rows and columns**             
+The DataGrid renders all rows and columns once the data is loaded. To optimize performance, set the [rowRenderingMode](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/scrolling/rowRenderingMode.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/scrolling/#rowRenderingMode') and [columnRenderingMode](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/scrolling/columnRenderingMode.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/scrolling/#columnRenderingMode') properties in the **scrolling** object to *"virtual"*. These properties specify that UI elements are only rendered when they come into the viewport. Use virtual row rendering when [paging is disabled](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/paging/enabled.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/paging/#enabled') or the [pageSize](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/paging/pageSize.md '/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/paging/#pageSize') is more than the default value. Virtual column rendering can be used when there are more than 20 columns outside the viewport.
```

Comment:
Add information about side effects of enabling this options. These options improve render performance after first render and after data operations but can decrease performance during scrolling because content is rendered during scrolling. The renderAsync global and column options can improve performance during scrolling.